### ISSUE TITLE
16-bit tiff file

### ISSUE THREAD

**Issue by clement-analogue** [DawyD/bm3d-gpu#2]
This project is not working with 16-bit tiff images. It is still doing something and the output is a 8-bit tiff pretty close to a random image.

**Comment by DawyD** [comment]
Yes, the architecture now works only with 8-bit images. The extension to 16-bit images is possible, but it would require substantial changes in the CUDA code as it was highly optimized for the memory consumption on the 8-bit images. Such change would therefore lead to significantly slower processing.
